library/src/interactor_impl.cxx

```diff
@@ -71,8 +75,102 @@ class interactor_impl::internals
 #endif
   }
 
+  //----------------------------------------------------------------------------
+  // Method defined to normalize the Z axis so all models are treated temporarily
+  // as Z-up axis models.
+  void zUpTransforms(vtkTransform* to, vtkTransform* from)
+  {
+    vtkRenderer* renderer =
+      this->VTKInteractor->GetRenderWindow()->GetRenderers()->GetFirstRenderer();
+    const double* up = renderer->GetEnvironmentUp();
+    const double* right = renderer->GetEnvironmentRight();
+    double fwd[3];
+    vtkMath::Cross(right, up, fwd);
+    const double m[16] = {
+      right[0], right[1], right[2], 0, //
+      fwd[0], fwd[1], fwd[2], 0,       //
+      up[0], up[1], up[2], 0,          //
+      0, 0, 0, 1,                      //
+    };
+    to->SetMatrix(m);
+    from->SetMatrix(m);
+    from->Inverse();
+  }
+
+  //----------------------------------------------------------------------------
+  // Set the view orbit position on the viewport.
+  enum class ViewType
+  {
+    VT_FRONT,
+    VT_BACK,
+    VT_RIGHT,
+    VT_LEFT,
+    VT_TOP,
+    VT_ISOMETRIC
+  };
+  static void SetViewOrbit(ViewType view, internals* self)
+  {
+    vtkNew<vtkTransform> toZup, fromZup;
+    self->zUpTransforms(toZup, fromZup);
+    camera& cam = self->Window.getCamera();
+    vector3_t up = { 0, 0, 1 };
+    point3_t pos = cam.getPosition();
+    point3_t foc = cam.getFocalPoint();
+    point3_t Axis, newPos;
+
+    /* convert coords to +Z up */
+    toZup->TransformPoint(pos.data(), pos.data());
```

Comment:
You will have a matrix3x3 instead of a transform here. Just replace `->TransformPoint` by `->MultiplyPoint`